lib/index.js

```diff
@@ -21,11 +21,10 @@ module.exports = class DatadogTransport extends Transport {
       throw new Error('Missing required option: `apiKey`')
     }
     this.opts = opts
-    if (this.opts.intakeRegion === 'eu') {
-      this.api = `https://http-intake.logs.datadoghq.eu/v1/input/${opts.apiKey}`
-    } else {
-      this.api = `https://http-intake.logs.datadoghq.com/v1/input/${opts.apiKey}`
-    }
+
+    const tld = opts.intakeRegion === 'eu' ? 'eu' : 'com'
+    const subdomain = opts.apiKey.startsWith('pub') ? 'browser-http-intake' : 'http-intake'
```

Comment:
Is there no better way to set the `subdomain`? Setting in based on the value of `apiKey` doesn't feel right.

Reply:
I am open to a separating the `apiKey`/`clientToken` opts, but the subdomains are dependent on the type of key you provide. I.e. sending a client token to `http-intake` does not result in logs

Reply:
@itsfadnis is separating apiKey / clientToken to determine the subdomain ok with you?

I'm interested in this feature